### PR TITLE
Reorganize the headers to isolate the nonstandard routines

### LINKED ISSUE #5: todo: figure out what to do regarding static inlines in the standard
Many of the macros in `pmix_defs.h` call static inline functions defined in the same header file. Those static inlines are not part of the PMIx Standard. Should the standard define these functions? Or should the macros using them also be promoted to functions like in pmix/pmix-standard#383.

Here is a list of the static inlines in `pmix_defs.h` today:
```c
static inline void* pmix_malloc(size_t n);
static inline void pmix_free(void *m);
static inline void* pmix_calloc(size_t n, size_t m);
static inline int pmix_argv_count(char **argv);
static inline pmix_status_t pmix_argv_append_nosize(char ***argv, const char *arg);
static inline pmix_status_t pmix_argv_prepend_nosize(char ***argv, const char *arg);
static inline pmix_status_t pmix_argv_append_unique_nosize(char ***argv, const char *arg);
static inline char **pmix_argv_split(const char *src_string, int delimeter);
static inline char *pmix_argv_join(char **argv, int delimiter);
static inline char **pmix_argv_copy(char **argv);
static inline pmix_status_t pmix_setenv(const char *name, const char *value, bool overwrite, char ***env);
static inline void pmix_darray_destruct(pmix_data_array_t *m);
static inline void pmix_value_destruct(pmix_value_t * m);
static inline void pmix_strncpy(char *dest, const char *src, size_t len);
static inline size_t pmix_keylen(const char *src);
static inline size_t pmix_nslen(const char *src);
```

### PR DESCRIPTION
**Pull request by jjhursey**
* Fixes #5
 * The non-standard supporting routines are in the `pmix_abi_support(_bottom).h`
   header to make them easily identified as non-standard.
 * Layer the headers to isolate the functions from the macros from the
   types and constants.
 * Add missing functions for
   * `PMIx_Disconnect`
   * `PMIx_Disconnect_nb`
   * `PMIx_Info_load`
   * `PMIx_Info_xfer`
 * Add a `pmix.h` that has the true function signatures

### CODE REVIEW

**Comment by jjhursey** [comment]
I'm not sure how we want to test this repo. I took care to make sure everything is ordered properly but didn't try to build anything against this. Any ideas?

**Comment by jjhursey** [comment]
FYI: I filed https://github.com/pmix/pmix-abi/issues/7 to discuss testing of this repo. https://github.com/pmix/pmix-abi/issues/1 is the place to talk about long-term maintenance and updating.

File: pmix_macros.h [hunk 1]
```diff
@@ -0,0 +1,1101 @@
...
+ * PMIx Standard types, constants, and callback functions
+ * *******************************************************************/
+#include "pmix_types.h"
+
+/* *******************************************************************
+ * The PMIx ABI Support functions that _do not_ rely on the macros defined
+ * in this header. They provide 'static inline' functions required to
+ * support these macro definitions.
+ * *******************************************************************/
+#include "pmix_abi_support_bottom.h"
```
Comment (raffenet) [inline pmix_macros.h:69]: Should this be `pmix_abi_support.h`? The "bottom" version is included at the end of this file.

Comment (jjhursey) [reply pmix_macros.h]: Good catch. Copy/paste issue. Repushed with the fix.

**Review by dsolt — APPROVED** [review]
These look good to me.  I have not yet tried to compile codes with them, but I will do that next.  The presentation/organization looks good.

**Comment by jjhursey** [comment]
Let's merge this and iterate a bit on master as we work out any details that crop up in broader testing.

**Pull request merged**